lib/phoenix_component.ex

```diff
@@ -1025,13 +1221,146 @@ defmodule Phoenix.Component do
     attrs
   end
 
-  defp raise_if_function_already_defined!(env, name, attrs) do
+  defp get_slots(module) do
+    Module.get_attribute(module, :__slots__) || []
+  end
+
+  defp pop_slots(env) do
+    slots =
+      env.module
+      |> get_slots()
+      |> Enum.reverse()
+
+    Module.delete_attribute(env.module, :__slots__)
+    slots
```

Comment:
I actually think that `delete_attribute` already return the attribute value, so maybe this could be written as:

```suggestion
    slots = Module.delete_attribute(env.module, :__slots__) || []
    Enum.reverse(slots)
```

If this is the case, can you please update `pop_attrs` too?